tests/unittests/test_sync.py

```diff
@@ -105,6 +105,43 @@ def test_sync_only_mid_child(self, mock_full_table, mock_incremental, mock_write
         self.assertEqual(mock_write_schemas.mock_calls[0], mock.call("projects", mock.ANY, mock.ANY))
         self.assertEqual(mock_write_schemas.mock_calls[1], mock.call("teams", mock.ANY, mock.ANY))
 
+    @mock.patch("tap_github.sync.get_stream_to_sync", return_value = [])
+    @mock.patch("tap_github.sync.get_selected_streams", return_value = [])
+    @mock.patch("tap_github.sync.update_currently_syncing_repo")
+    def test_no_streams_selected(self, mock_update_curr_sync, mock_selected_streams, mock_sync_streams, 
+                                 mock_incremental, mock_write_schemas, mock_write_state):
+        """
+        Test if no streams are selected then the state does not update,
+        and `update_currently_syncing_repo` function is not called.
+        """
+
+        state = {
+                    "currently_syncing_repo": "singer-io/test-repo",
+                    "bookmarks": {},
+                    "currently_syncing": "teams"
+                }
```

Comment:
Use `state = expected_state = {...}` so it get clear that state is not changing.